native/core/src/execution/planner.rs

```diff
@@ -830,6 +830,25 @@ impl PhysicalPlanner {
                 ));
                 Ok(array_has_any_expr)
             }
+            ExprStruct::ArrayCompact(expr) => {
```

Comment:
Wondering if we can use https://github.com/apache/datafusion-comet/blob/main/spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala#L2116
`scalarExprToProtoWithReturnType` or `scalarExprToProto`
so that we do not have to define here for every Array related function

Reply:
I can try to do it, but is it necessary to do it as part of this issue?

Reply:
done.